packages/gestalt/src/Badge.tsx

```diff
@@ -71,52 +82,123 @@ export default function Badge({ position = 'middle', text, type = 'info', toolti
     'warning': 'workflow-status-warning',
     'success': 'check-circle',
     'neutral': 'lock',
-    'recommendation': 'performance-plus',
+    'recommendation': 'sparkle',
     'darkWash': 'info-circle',
     'lightWash': 'info-circle',
   });
 
+  const COLOR_ICON_MAP = Object.freeze({
+    'info': 'info',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
+  const COLOR_TEXT_MAP = Object.freeze({
+    'info': 'shopping',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
   let styleType: TypeOptions | InteractiveTypeOptions = type;
 
   if (shouldUseTooltip) {
     styleType = `interactive-${type}`;
   }
 
-  const csBadge = cx(styles.Badge, styles[position], styles[styleType]);
+  const { handleOnBlur, handleOnFocus, handleOnMouseEnter, handleOnMouseLeave, isFocused } =
+    useInteractiveStates();
+
+  const cxStyles = useMemo(
+    () =>
+      cx(styles.badge, styles[styleType], {
+        [styles.middle]: !shouldUseTooltip && position === 'middle',
+        [styles.top]: !shouldUseTooltip && position === 'top',
+        [styles.focusInnerBorder]:
+          isInVRExperiment &&
+          isFocused &&
+          isFocusVisible &&
+          !['darkWash', 'lightWash'].some((color) => color === type),
+        [styles.focusInnerBorderLight]:
+          isInVRExperiment && isFocused && isFocusVisible && type === 'darkWash',
+        [styles.focusInnerBorderDark]:
+          isInVRExperiment && isFocused && isFocusVisible && type === 'lightWash',
+      }),
+    [isFocused, isInVRExperiment, styleType, position, shouldUseTooltip, isFocusVisible, type],
+  );
+
+  const cxPositionStyles = useMemo(
```

Comment:
How come this can't be merged with the above?

Reply:
cos TapArea is outside the Badge.

we need it splitted